src/nrepl/jack-in.ts

```diff
@@ -147,26 +175,43 @@ export async function calvaJackIn() {
 
 export async function calvaDisonnect() {
 
-    if (state.deref().get('connected')) {
+    if (utilities.getConnectedState()) {
         calvaJackout();
         connector.default.disconnect();
         return;
+    } else if (utilities.getConnectingState() ||
+        utilities.getLaunchingState()) {
+        vscode.window.showInformationMessage(
+            "Do you want to interupt the connection process?",
+            { modal: true },
+            ...["Ok"]).then((value) => {
+                if (value == 'Ok') {
+                    const outputChannel = state.outputChannel();
+                    calvaJackout();
+                    connector.default.disconnect();
+                    outputChannel.appendLine("Interrupting Jack-in process.");
+                    outputChannel.show();
+                }
+            });
+        return;
     }
     vscode.window.showInformationMessage("Not connected to a REPL server");
 }
 
 export async function calvaJackInOrConnect() {
 
     let commands = {};
-    if (!state.deref().get('connected')) {
+    if (!utilities.getConnectedState() &&
+        !utilities.getConnectedState() &&
+        !utilities.getLaunchingState()) {
```

Comment:
There seems to be the same statement two times here `!utilities.getConnectedState()`